### DISCUSSION TITLE
Next-Auth Hasura and Next JS

### DISCUSSION THREAD

**Discussion by rodallanmac** [nextauthjs/next-auth#934] [Help]
**HELP**
Awesome experience with using NextAuthjs with your favourite auth provider - thank you all.

Does anyone have a repo that can authenticate a local:3000 to a Hasura postgres graphql instance that is running on HEROKU. (ie: not a local hasura instance via a local docker deployment as per @SharadKumar and @ghoshnirmalya)

It seems that Hasura only offers Auth0 and Firebase Auth jwt tokens to be authorised through their custom jwt claims.

Happy to collaborate.

**Comment by arvigeus** [comment]
Hi! I don't know if this would help: https://github.com/arvigeus/react-modern-state/blob/master/src/server/auth.ts

**Comment by ghoshnirmalya** [comment]
@rodallanmac You can use [Hasura Next.js Boilerplate](https://github.com/ghoshnirmalya/nextjs-hasura-boilerplate) with a Hasura Postgres GraphQL instance by modifying the correct credentials in the [frontend/.env.example](https://github.com/ghoshnirmalya/nextjs-hasura-boilerplate/blob/master/frontend/.env.example#L1-L6) file. You need to do that before running the [**yarn bootstrap**](https://nirmalyaghosh.com/guides/nextjs-hasura-boilerplate/04-getting-started#installation) command.

Also, you can only run the Next.js application.

**Reply by Manubi** [reply]
@ghoshnirmalya Could you write a tutorial how you made the setup? The boilerplate helps but it doesn't answer a lot of questions...
Would really appreciate it! Thanks!

**Reply by brunocrosier** [reply]
@Manubi I don't know if you're still looking for an answer on this, but I'm currently hacking together a setup to get Hasura (Cloudf / free tier) + Heroku Postgres + next-auth all working together (basically trying to create the ultimate free fullstack setup with auth!). It's heavily inspired by @ghoshnirmalya 's repo but I suck at docker so I wanted to do it with Hasura's cloud version.

I've pretty much got everything working now, except for Apollo on non-SSR pages. Shoot me an email if you wanna see where I'm at or bounce ideas about.

**Reply by joeyorjohn** [reply]
@brunocrosier did you ever happen to finish this? Pulling my hair out rn haha

**Reply by brunocrosier** [reply]
@joeyorjohn sorry mate I think I ended up giving up eventually 😂

What are you struggling with? Feel free to email me, my address is on my profile

**Comment by iaincollins** [comment]
In addition to the example project from @ghoshnirmalya there is also a tutorial made by the Hasura folks!

https://hasura.io/blog/next-js-jwt-authentication-with-next-auth-and-integration-with-hasura/ which includes an example repo.

They have also made a YouTube video showing how to use Hasura with NextAuth.js:

https://www.youtube.com/watch?v=Vy0UldnCmrw&feature=emb_title

**Comment by rodallanmac** [comment]
I know this is a couple of months late but a big shout out to Praveen for
the excellent integration of Hasura jwt and next-auth. All questions
answered.
